### ISSUE TITLE
How does local peering work?

### ISSUE THREAD

**Issue by gitfamhub** [yggdrasil-network/yggdrasil-go#418]
I've tried cjdns before and while it worked well when peering over the Internet, I never quite understood how local peering worked. Let's say my router is running Yggdrasil (waiting on that OpenWrt build ;_;) and I want to peer with another router a few blocks away, how would I go about that?

Libremesh has [good documentation](https://libremesh.org/docs/en_connecting_nodes.html) on this, but I can't find any for Yggdrasil. Much appreciated!

**Comment by neilalexander** [comment]
Yggdrasil sends out, and listens for, multicast beacons which allow nodes on the same subnet to discover each other for peering. This pretty much works on any link which supports multicast IPv6, including point-to-point Ethernet radio links. 

In this case, you could have a router and a wireless station in each building, run Yggdrasil on each router, enable multicast peerings on the interface that connects the router to the wireless station. The two nodes should automatically peer over the air.

**Comment by gitfamhub** [comment]
Does this work even when they're password-protected?

**Comment by neilalexander** [comment]
Yggdrasil doesn’t have password protection on peerings like cjdns does. We can optionally whitelist public keys for remote connectivity. 

Multicast-discovered peerings are always allowed and take priority over the whitelist.

**Comment by gitfamhub** [comment]
Thanks, but, I meant if the routers/networks are password-protected, not the peers.

**Comment by neilalexander** [comment]
If you are using WPA then only devices who know the WPA passphrase will be able to connect and peer. For a point-to-point radio link, both ends of the wireless link would need the same WPA passphrase.

**Comment by Arceliar** [comment]
Yggdrasil doesn't see any difference between a wifi and a wired network device, as long as they support link-local multicast. Wifi looks the same as ethernet as far as we're concerned.

If you're trying to use it as a mesh network, then you want to configure the wifi in IBSS ("adhoc") or 802.11s ("mesh point") mode. I'm told that 802.11s is better if your hardware supports it, but few things do, so IBSS is more likely to work. I've never gotten either to work successfully with WPA, but I only tried as a matter of curiosity, so maybe I just overlooked something.

If you only have 2 devices to connect, such as with directional point-to-point links, then it's OK to have one device run as a normal wifi access point, and have the other device connect in client/station mode. This is less flexible, but I would expect it to work a lot better on most hardware.
